src/server.py

```diff
@@ -73,10 +131,17 @@ def get_relation_sample():
     selection_conds = data["selection_conds"]
     groupby_conds = data.get("groupby_conds") or []
     orderby_conds = data.get("orderby_conds") or []
-    limit = data.get("limit", 100)
+    custom_order_pref = data.get("custom_order_pref") or []
+    limit = data.get("limit", 1000)
     # Return the sample data
-    return jsonify(
-        dashboard.get_relation_sample(relation, selection_conds, groupby_conds, orderby_conds, agg_exprs, limit))
+    temp = dashboard.get_relation_sample(relation, selection_conds, groupby_conds, orderby_conds, agg_exprs, limit,
+                                      custom_order_pref)
+    # We need a better way to handle NaNs while converting to JSON
+    d = simplejson.dumps(temp, ignore_nan=True, default=datetime.isoformat)
+    response = app.response_class(
+        response=d,
+        status=200,mimetype='application/json')
+    return response
```

Comment:
I don't follow why we need simple json

Reply:
Python's json lib outputs NaN in the json output instead of null. NaN is not valid json so th e front end can't understand or parse it.
simplejson handles it well, instead of using simplejson, we can also do a simple text.replace("NaN", "null"), but that's not reliable if NaN is part of actual text columns

https://stackoverflow.com/questions/28639953/python-json-encoder-convert-nans-to-null-instead